modules/repository-url/build.gradle

```diff
@@ -39,31 +39,38 @@ restResources {
  // This directory is shared between two URL repositories and one FS repository in YAML integration tests
 File repositoryDir = new File(project.buildDir, "shared-repository")
 
-/** A task to start the URLFixture which exposes the repositoryDir over HTTP **/
-def urlFixtureTaskProvider = tasks.register("urlFixture", AntFixture) {
-  dependsOn "testClasses"
-  doFirst {
+def testJar = tasks.register("testJar", Jar) {
+  from sourceSets.test.output
+  classifier = "test"
+}
+
+tasks.named("preProcessFixture").configure {
+  inputs.files(testJar, sourceSets.test.runtimeClasspath)
+  doLast {
+    file("${testFixturesDir}/shared").mkdirs()
+    project.copy {
+      from testJar.get().archiveFile
```

Comment:
Is gradle smart enough to do `from testJar` here since `testJar` is a provider of an `AbstractArchiveTask`? It doesn't change anything in terms of task realization but it's just more concise.

Reply:
it is. fixed